Improve Parca dashboard

mmkay opened the pull request:
## Issue
<!-- What issue is this PR trying to solve? -->
The dashboard added to the charm wasn't great yet. There are limits to how much we can do based on the available metrics, but this is closer to what we have with tempo.

## Solution
<!-- A summary of the solution addressing the above issue -->
Update dashboard to the latest upstream version and then add more relevant panels.

## Context
<!-- What is some specialized knowledge relevant to this project/technology -->
We can't expose the Parca version on the dashboard due to https://github.com/parca-dev/parca/issues/3645 .

## Testing Instructions
<!-- What steps need to be taken to test this PR? -->
Either deploy this PR or see the following screenshots:

![Screenshot from 2025-01-24 15-25-27](https://github.com/user-attachments/assets/0cfc921a-d376-4339-93c5-4c6c317dc298)
![Screenshot from 2025-01-24 15-25-45](https://github.com/user-attachments/assets/6fd8aab4-f54f-4b1e-a09a-9c50a99753ff)
![Screenshot from 2025-01-24 15-25-52](https://github.com/user-attachments/assets/c2129a60-995f-4ea8-9bbb-a43757c15388)


## Upgrade Notes
<!-- To upgrade from an older revision of charmed prometheus, ... -->

michaeldmitry reviewed (commented):
All looks fine except for a couple of things I have questions for:

1- The `Scraped Targets` panel is misleading to me
I thought it would give you how many targets you're currently scraping.
But,
a- The metric used is a counter not a gauge
b- I think the metrics is not giving us what we think it does
![image (8)](https://github.com/user-attachments/assets/dd43284f-7ee0-4779-af0d-8d12d1eb9be6)
Its showing 3 although I'm only scraping 1 target.

2- The `Profiles storage size` panel
This panel uses the metric for `frostDB` which IIUC, is the in-memory storage parca is using. So, if we have profiles persistence, the values this panels will show would periodically drop at some point, right? I think if we rename it to `In-memory profiles stoarge size` it would better describe the excepted behaviour of this panel.

3- The `Samples Appended` panel needs a better title 
I thought its a counter-type of panel that would tell you how many samples have you scraped, but its actually how many samples are  being inserted per second.

4- `Write from parca-agent`
Current;y, yes, only parca-agent pushes profiles to parca, but in theory, anything can push to parca. Should we rename it to encapsulate the general behaviour?

mmkay commented:
@michaeldmitry thanks, updated the dashboard to reflect that. For 1), I've dropped the panel altogether, there doesn't seem to be a clear enough metric that would show that. Let's iterate on it further once parca metrics evolve :)
